### PR TITLE
add macro fo English prep usage

### PR DESCRIPTION
**Pull request by ablodge**

### CODE REVIEW

**Comment by nschneid** [comment]
OK nice.

Brainstorming on the syntax: how about `[p en/in Locus--Locus]` for linking to the usage and `[p en/in]` for linking to the preposition itself? With `p` as the macro name we wouldn't need to recreate it for each language. Is this possible within the constraints of the macro system?

Also, for language-specific documentation pages, it would be nice if the language did not have to be specified. E.g., on the page /en/in, `[p near]` to link to /en/near (note that `[[near]]` always links to /near), and `[p near Locus--Locus]` to link to /en/near/Locus--Locus.

For completeness, it's probably also worth having macros for supersenses and construals. Maybe `[ss Locus]` and `[ss Locus--Locus]`. The links would be equivalent to using `[[...]]`, but the macro would let us apply custom styling.

**Comment by ablodge** [comment]
We can definitely do some of that. 

I think argument names are required, e.g. `[p p:near c:Locus--Locus]`, but it doesn't look hard to overwrite without the arg names. 

I'm not sure if macros can access the page location, so it might be hard to have any language-specific defaults. Do you know if there's a way to get the page directory or other metadata in python?

**Comment by nschneid** [comment]
> I'm not sure if macros can access the page location, so it might be hard to have any language-specific defaults. Do you know if there's a way to get the page directory or other metadata in python?

There must be a place where the wiki software calls the Markdown renderer. I bet it has access to the URL.

File: wiki/plugins/macros/mdx/macro.py [hunk 1]
```diff
@@ -105,11 +132,24 @@ def wikilink(self):
...
-        short_description=_('Link to English Adposition Usage'),
-        help_text=_('Create a link to a preposition-construal pair'),
-        example_code='[enprep prep:in construal:Locus--Locus]',
+    def p(self, *args):
+        if len(args)>1:
+            prep, construal = args[0], args[1]
+            return '['+prep+'](/'+prep+'/'+construal+')'
+        return '['+args[0]+'](/'+args[0]+')'
+    p.meta = dict(
+        short_description=_('Link to English Adposition, Usage'),
```
Comment (nschneid) [inline wiki/plugins/macros/mdx/macro.py]: remove "English"

**Comment by nschneid** [comment]
Cool! Have you tested this by running the server locally?

**Pull request closed without merging**